### PR TITLE
Add nuxeo image

### PR DESCRIPTION
**Pull request by dmetzler**
Hello,

We would like to provide an official Nuxeo Docker image.

Thx for reviewing.

### CODE REVIEW

**Comment by Moghedrin** [comment]
LGTM

**Comment by tianon** [comment]
I've made a few minor recommendations in https://github.com/nuxeo/docker-nuxeo/pull/1, but otherwise this is looking pretty good (understanding of course that the massive bulk of LibreOffice is kind of unavoidable here). :+1:

**Comment by dmetzler** [comment]
I found a problem in the way CLID and packages are installed.

**Comment by tianon** [comment]
Would you be willing to update `test/config.sh` to add a new `nuxeo` section as well so we can test your test while we build-test this PR? :smile: (https://github.com/docker-library/official-images/blob/fbfba6b77d868f62c1a8d095929974db94ee513c/test/config.sh#L88-L89)

**Comment by tianon** [comment]
Content LGTM :+1:

File: test/tests/nuxeo-basics/run.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+NUXEO_TEST_TRIES=10
+
+cname="nuxeo-container-$RANDOM-$RANDOM"
+cid="$(docker run -d --name "$cname" "$image")"
+trap "docker rm -vf $cid > /dev/null" EXIT
+
+get() {
+	docker run --rm -i \
+		--link "$cname":nuxeo \
+		"$image" curl -s \
```
Comment (md5) [inline test/tests/nuxeo-basics/run.sh]: It might be good to throw a `-f` on this `curl` command.

Comment (tianon) [inline test/tests/nuxeo-basics/run.sh]: Good catch! :+1:

**Comment by tianon** [comment]
LGTM

Build test of #1223; 6d5ee3c (`nuxeo`):

```console
$ bashbrew build "nuxeo"
Cloning nuxeo (git://github.com/nuxeo/docker-nuxeo) ...
Processing nuxeo:latest ...
Processing nuxeo:LTS-2015 ...
Processing nuxeo:LTS ...
Processing nuxeo:7.10 ...
Processing nuxeo:7 ...
Processing nuxeo:7.4 ...
Processing nuxeo:LTS-2014 ...
Processing nuxeo:6 ...
Processing nuxeo:6.0 ...
$ bashbrew list --uniq "$url" | xargs test/run.sh
testing nuxeo:latest
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'nuxeo-conf' [5/6]...failed
	'nuxeo-basics' [6/6].../usr/src/pr/test/tests/nuxeo-basics/run.sh: line 24: ./../../retry.sh: No such file or directory
failed
testing nuxeo:7.4
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'nuxeo-conf' [5/6]...failed
	'nuxeo-basics' [6/6].../usr/src/pr/test/tests/nuxeo-basics/run.sh: line 24: ./../../retry.sh: No such file or directory
failed
testing nuxeo:LTS-2014
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'nuxeo-conf' [5/6]...failed
	'nuxeo-basics' [6/6].../usr/src/pr/test/tests/nuxeo-basics/run.sh: line 24: ./../../retry.sh: No such file or directory
failed
```

There is at least one failure in the above build log.

(Looks like the tests need some work, but the image is solid, so I plan to merge and we can fix the tests in a follow-up or two. :+1:)

**Comment by dmetzler** [comment]
Ok, I'll check the test. I'll try to build a correct environment, since it doesn't run on OSX. 
All in all this is a good christmas gift ;-)

**Comment by tianon** [comment]
No worries -- we're working on getting the tests at least runnable on OSX (they're pretty close right now) by using `docker build` more aggressively (instead of bind mounts) etc so that they can at least run inside `dind`, but the wrapper script we've got (`test-pr.sh`) still can't handle OSX yet, although it should be pretty close.

I'm taking a look at the tests now also -- I'm not sure what was wrong with `nuxeo-conf`; I run it now with just whitespace changes in the script and it passes. :confused:  Working on `nuxeo-basics` now too. :+1:

**Comment by tianon** [comment]
Here's the money shot with a few minor tweaks to the test:
```console
...
15:34:47,200 INFO  [AjpProtocol] Starting ProtocolHandler ["ajp-bio-0.0.0.0-8009"]
15:34:47,200 INFO  [Catalina] Server startup in 26350 ms
Dec 24, 2015 3:34:48 PM com.sun.jersey.server.impl.application.WebApplicationImpl _initiate
INFO: Initiating Jersey application, version 'Jersey: 1.18.3 12/01/2014 08:23 AM'
Dec 24, 2015 3:34:48 PM com.sun.jersey.spi.inject.Errors processErrorMessages
WARNING: The following warnings have been detected with resource and/or provider classes:
  WARNING: A HTTP GET method, public java.lang.String org.nuxeo.elasticsearch.http.readonly.Main.searchWithPayload(java.lang.String,javax.ws.rs.core.UriInfo,javax.ws.rs.core.MultivaluedMap) throws java.io.IOException,org.json.JSONException, should not consume any entity.
  WARNING: A HTTP GET method, public java.lang.String org.nuxeo.elasticsearch.http.readonly.Main.searchWithPayload(javax.ws.rs.core.UriInfo,javax.ws.rs.core.MultivaluedMap) throws java.io.IOException,org.json.JSONException, should not consume any entity.
  WARNING: A HTTP GET method, public java.lang.String org.nuxeo.elasticsearch.http.readonly.Main.searchWithPayload(java.lang.String,java.lang.String,javax.ws.rs.core.UriInfo,javax.ws.rs.core.MultivaluedMap) throws java.io.IOException,org.json.JSONException, should not consume any entity.
15:34:49,128 WARN  [WebEngineExceptionMapper] Exception in JAX-RS processing
org.nuxeo.ecm.webengine.model.TypeNotFoundException: Type not found: default-domain
	at org.nuxeo.ecm.webengine.model.impl.ModuleImpl.getType(ModuleImpl.java:197)
	at org.nuxeo.ecm.webengine.model.impl.AbstractWebContext.newObject(AbstractWebContext.java:312)
	at org.nuxeo.ecm.webengine.model.impl.AbstractResource.newObject(AbstractResource.java:232)
	at org.nuxeo.ecm.restapi.server.jaxrs.RepositoryObject.route(RepositoryObject.java:87)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.dispatch(SubLocatorRule.java:197)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.invokeSubLocator(SubLocatorRule.java:183)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:110)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.SubLocatorRule.accept(SubLocatorRule.java:137)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1542)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1473)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1419)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1409)
	at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:409)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:558)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:733)
	at org.nuxeo.ecm.webengine.app.jersey.WebEngineServlet.containerService(WebEngineServlet.java:171)
	at org.nuxeo.ecm.webengine.app.jersey.WebEngineServlet.service(WebEngineServlet.java:148)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.webengine.jaxrs.session.SessionCleanupFilter.run(SessionCleanupFilter.java:43)
	at org.nuxeo.ecm.webengine.jaxrs.HttpFilter.doFilter(HttpFilter.java:42)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.webengine.app.WebEngineFilter.doFilter(WebEngineFilter.java:92)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.webengine.jaxrs.context.RequestContextFilter.run(RequestContextFilter.java:41)
	at org.nuxeo.ecm.webengine.jaxrs.HttpFilter.doFilter(HttpFilter.java:42)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.core.management.jtajca.internal.Log4jWebFilter.doFilter(Log4jWebFilter.java:67)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.ui.web.auth.NuxeoAuthenticationFilter.doFilterInternal(NuxeoAuthenticationFilter.java:580)
	at org.nuxeo.ecm.platform.ui.web.auth.service.NuxeoAuthFilterChain.doFilter(NuxeoAuthFilterChain.java:36)
	at org.nuxeo.ecm.platform.ui.web.auth.oauth.NuxeoOAuthFilter.doFilter(NuxeoOAuthFilter.java:120)
	at org.nuxeo.ecm.platform.ui.web.auth.service.NuxeoAuthFilterChain.doFilter(NuxeoAuthFilterChain.java:34)
	at org.nuxeo.ecm.platform.ui.web.auth.oauth2.NuxeoOAuth2Filter.doFilter(NuxeoOAuth2Filter.java:68)
	at org.nuxeo.ecm.platform.ui.web.auth.service.NuxeoAuthFilterChain.doFilter(NuxeoAuthFilterChain.java:34)
	at org.nuxeo.ecm.platform.ui.web.auth.NuxeoAuthenticationFilter.doFilter(NuxeoAuthenticationFilter.java:393)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.web.common.exceptionhandling.NuxeoExceptionFilter.doFilter(NuxeoExceptionFilter.java:75)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.web.common.encoding.NuxeoEncodingFilter.doFilter(NuxeoEncodingFilter.java:73)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:748)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:486)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:411)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:338)
	at org.nuxeo.ecm.restapi.server.APIServlet.service(APIServlet.java:46)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.web.common.requestcontroller.filter.NuxeoThreadTrackerFilter.doFilter(NuxeoThreadTrackerFilter.java:28)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.core.management.jtajca.internal.Log4jWebFilter.doFilter(Log4jWebFilter.java:67)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.ui.web.rest.FancyURLFilter.doFilter(FancyURLFilter.java:118)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.web.common.requestcontroller.filter.NuxeoRequestControllerFilter.doFilter(NuxeoRequestControllerFilter.java:140)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.webdav.service.WIRequestFilter.doFilter(WIRequestFilter.java:59)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.web.common.requestcontroller.filter.NuxeoCorsFilter.doFilter(NuxeoCorsFilter.java:49)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.nuxeo.ecm.platform.web.common.exceptionhandling.NuxeoExceptionFilter.doFilter(NuxeoExceptionFilter.java:75)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
...
```

**Comment by tianon** [comment]
Ah, it's missing `path/`! :smile:

**Comment by dmetzler** [comment]
yes, I was just sending the fix ;-) but if you get it it's ok.

**Comment by tianon** [comment]
Heh, I don't mind letting you take over for the PR one bit!  Here's what I've got: (a few minor whitespace changes for readability/consistency as well)

```diff
diff --git a/test/tests/nuxeo-basics/run.sh b/test/tests/nuxeo-basics/run.sh
index e3f3bf8..cbfc5bd 100755
--- a/test/tests/nuxeo-basics/run.sh
+++ b/test/tests/nuxeo-basics/run.sh
@@ -15,23 +15,41 @@ trap "docker rm -vf $cid > /dev/null" EXIT
 get() {
 	docker run --rm -i \
 		--link "$cname":nuxeo \
-		"$image" curl -fs \
-		-H "Content-Type:application/json" \
-		-u Administrator:Administrator \
-		http://nuxeo:8080/nuxeo/api/v1/$1 
+		--entrypoint curl \
+		"$image" \
+		-fs \
+			-H "Content-Type:application/json" \
+			-u Administrator:Administrator \
+			"http://nuxeo:8080/nuxeo/api/v1/$1"
 }
 
-. "./../../retry.sh" --tries "$NUXEO_TEST_TRIES" \
-	--sleep "$NUXEO_TEST_SLEEP" \
-	"get default-domain/workspaces"
+python() {
+	docker run --rm -i \
+		--entrypoint python3 \
+		"$image" \
+		"$@"
+}
 
-PATH1="default-domain/workspaces"
+PATH1='default-domain/workspaces'
+
+. "$dir/../../retry.sh" \
+	--tries "$NUXEO_TEST_TRIES" \
+	--sleep "$NUXEO_TEST_SLEEP" \
+	"get 'path/$PATH1'"
 
 # First get a document by its path to get its id
-DUID=$(get path/$PATH1 | python -c 'import json,sys;obj=json.load(sys.stdin);print obj["uid"];')
+DUID="$(get "path/$PATH1" | python -c '
+	import json, sys;
+	obj = json.load(sys.stdin);
+	print obj["uid"];
+')"
 
 # Then get the same document by its id
-PATH2=$(get id/$DUID | python -c 'import json,sys;obj=json.load(sys.stdin);print obj["path"];')
+PATH2="$(get "id/$DUID" | python -c '
+	import json, sys;
+	obj = json.load(sys.stdin);
+	print obj["path"];
+')"
 
 # Compare both path
-[ "/$PATH1" == "$PATH2" ]
+[ "/$PATH1" = "$PATH2" ]
diff --git a/test/tests/nuxeo-conf/run.sh b/test/tests/nuxeo-conf/run.sh
index 09583ca..f7af3c4 100755
--- a/test/tests/nuxeo-conf/run.sh
+++ b/test/tests/nuxeo-conf/run.sh
@@ -1,7 +1,6 @@
 #!/bin/bash
 set -eo pipefail
 
-
 image="$1"
 
 export NUXEO_DEV_MODE='true'
@@ -11,5 +10,4 @@ docker run --rm -i \
 	-e NUXEO_DEV_MODE \
 	-e NUXEO_AUTOMATION_TRACE \
 	"$image" \
-	./bin/nuxeoctl showconf | grep org.nuxeo 
-
+	nuxeoctl showconf | grep ^org.nuxeo
diff --git a/test/tests/remove-onbuild.sh b/test/tests/remove-onbuild.sh
index 8fb81fb..c0f17e1 100755
--- a/test/tests/remove-onbuild.sh
+++ b/test/tests/remove-onbuild.sh
@@ -47,8 +47,10 @@ mkdir -p "$tmp/$newId"
 echo "$jsonString" > "$tmp/$newId/json"
 echo -n '1.0' > "$tmp/$newId/VERSION"
 dd if=/dev/zero of="$tmp/$newId/layer.tar" bs=1k count=1 &> /dev/null # empty tar file
+
 cat > "$tmp/repositories" <<EOF
 {"$outImage":{"$outTag":"$newId"}}
 EOF
+
 docker rmi -f "$out" &> /dev/null || true # avoid "already exists, renaming the old one" from "docker load"
 tar -cC "$tmp" . | docker load
```

**Comment by tianon** [comment]
Doh scratch that python string whitespace, it breaks the script heh

**Comment by dmetzler** [comment]
I don't mind either. Feel free to make it, it seems you have more fixes than me and moreover I have to leave and don't know when I'll be abel to commit again on that work (going off preparing XMas eve)

**Comment by tianon** [comment]
Heh, ok -- have a nice holiday!  I'll get this dialed in and on the flip side `nuxeo` should be pushed soon! (it's finished building on the official server) :christmas_tree:

You did all the hard work already; this is the stuff I can handle without too much research. :smile:

**Pull request merged**